openedx/core/djangoapps/course_live/views.py

```diff
@@ -207,3 +215,94 @@ def get_provider_data(course_id: str) -> Dict:
                 "available": AVAILABLE_PROVIDERS
             }
         }
+
+
+class CourseLiveIframeView(APIView, LtiCourseLaunchMixin):
+    """
+    A view for retrieving course live iFrame.
+
+    Path: ``api/course_live/iframe/{course_id}/``
+
+    Accepts: [GET]
+
+    ------------------------------------------------------------------------------------
+    GET
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a course live zoom iframe.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the course.
+        * 404: The requested course does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will be iframe HTML.
+
+    **Example**
+
+        {
+            "iframe": "
+                        <iframe
+                            id='lti-tab-embed'
+                            style='width: 100%; min-height: 800px; border: none'
+                            srcdoc='{srcdoc}'
+                            >
+                        </iframe>
+                        ",
+        }
+
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (permissions.IsAuthenticated, IsEnrolled)
+
+    ROLE_MAP = {
+        'student': 'Student',
+        'staff': 'Administrator',
+    }
+
+    def _get_lti_config(self, course: CourseBlock) -> LtiConfiguration:
+        """
+        Get course live configurations
+        """
+        return CourseLiveConfiguration.get(course.id).lti_configuration
+
+    def _get_lti_roles(self, user: AbstractBaseUser, course_key: CourseKey) -> str:
+        return self.ROLE_MAP.get(
+            get_user_role(user, course_key),
+            self.DEFAULT_ROLE,
+        )
+
+    @ensure_valid_course_key
+    @verify_course_exists()
+    def get(self, request, course_id: str, **_kwargs) -> Response:
+        """
+        Handle HTTP/GET requests
+        """
+        course_key = CourseKey.from_string(course_id)
+
```

Comment:
Please try `CourseLiveTab().render_to_fragment(request,course).content` and remove `LTICourseLaunchMixin`

Reply:
I'm overloading `_get_lti_roles` as @asadazam93 told me that there's a discussion specific role that we don't want to include here. That's why inheritance from `LTICourseLaunchMixin`.